### ISSUE TITLE
incorrect check for running instance of ds9

### ISSUE THREAD

**Issue by ericmandel** [ericmandel/pyds9#96]
@DougBurke Whenever I start up pyds9, I can a long message beginning:
````
An instance of ds9 was found to be running before we could
start the 'xpans' name server. You will need to perform a
bit of manual intervention in order to connect this
existing ds9 to Python.
````
regardless of whether DS9 is running or not. This is because there is a too-general check on whether DS9 is running. In the ds9_xpans routine around line 61, we see:
````
            # start up xpans
            subprocess.Popen([_fname, "-e"])
            # if ds9 is already running, issue a warning
            p = subprocess.Popen(['ps', '-A'], stdout=subprocess.PIPE,
                                 universal_newlines=True)
            pslist = p.communicate()[0]   # get the std out
            if 'ds9' in pslist:
                print(tw.dedent("""
                    An instance of ds9 was found to be running before we could
````
Using ps is fine but the simple check for 'ds9' in plist is not. On my Mac, as soon as I start up xpans, I see this:
````
ps -A | egrep ds9
63140 ttys000    0:00.00 /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pyds9/xpans -e
````
and the fact that xpans is installed in the pyds9 directory guarantees that 'ds9' will be in pslist.

I think we need to filter out lines from the ps output that contain "xpans" ... which I don't know how to do offhand. Can you please help?

**Comment by DougBurke** [comment]
I think the tricky thing here is how general do you want to be: e.g.

 - does this have to work on windows too, or is there a different path for that?
 - how portable is `ps -A` between different UNIX-like systems

Although we can put together something that improves upon the current situation without  addressing either of these.

**Comment by ericmandel** [comment]
This check is probably overkill ... I would skip Windows happily, leave -A as is (it works on Linux and Mac) and just filter out instances related to pyds9. These instances almost certainly are the result of the newer install code beyond the primitive stuff I wrote a decade ago.

**Comment by DougBurke** [comment]
Oh. That's why I'm not seeing it. I'm using an out-of-tree version of `xpans` (so it's not triggering the problem).

**Comment by DougBurke** [comment]
@ericmandel - if you have the Aqua version of DS9 running, what does `ps -A` report for that line? Is it  `...SAOImageDS9.app` (maybe with a space) or something different?

**Comment by ericmandel** [comment]
It does the right thing:
````
eric% ps -A | egrep ds9 | egrep -v egrep
49510 ??         0:01.37 /Applications/SAOImageDS9.app/Contents/MacOS/ds9
````

**Comment by DougBurke** [comment]
Closed by #97
